public/gameInfo.php

```diff
@@ -89,6 +91,29 @@
 ?>
 <?php if ($gate): ?>
     <?php RenderContentStart($pageTitle) ?>
+    <script>
+        function disableMatureContentWarningPreference() {
+            const isLoggedIn = <?= isset($userDetails) ?>;
+            if (!isLoggedIn) {
+                throw new Error('Tried to modify settings for an unauthenticated user.');
+            }
+
+            const newPreferencesValue = <?= ($userDetails['websitePrefs'] ?? 0) | (1 << $matureContentPref) ?>;
+            const gameId = <?= $gameID ?>;
+
+            fetch('/request/user/update-notification.php', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                    'X-CSRF-TOKEN': document.querySelector('meta[name="csrf-token"]').getAttribute('content')
+                },
+                body: `preferences=${newPreferencesValue}`,
+                credentials: 'same-origin'
+            }).then(() => {
+                window.location = `/game/<?= $gameID ?>`;
+            })
```

Comment:
While I appreciate the modern approach, note that this escapes the prepared ajax http interceptor setup: https://github.com/RetroAchievements/RAWeb/blob/master/public/js/all.js#L9-L45

I have not yet found the time to further improve the frontend setup; it would include generalizing `fetch` usage in the application so we can replace jquery down the line, and move all.js to resources to be included in the build step.